src/timeout.ts

```diff
@@ -202,15 +196,29 @@ export abstract class TimeoutContext {
   abstract refreshed(): TimeoutContext;
 }
 
+function preventClear(timeout: Timeout | null): Timeout | null {
+  if (timeout == null) return null;
+  return Object.create(timeout, {
+    clear: {
+      value: function clear() {
+        // do nothing
+      }
+    }
+  });
+}
+
+export function isCSOTTimeoutContext(
```

Comment:
Having a method that narrows the base abstraction to a concrete subtype as a required method on the base abstraction breaks the base abstraction.  I've moved this to a floating function to keep the abstraction clean.